Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs

```diff
@@ -0,0 +1,109 @@
+using System;
+using UnityEngine;
+#if UNITY_2021_1_OR_NEWER
+using UnityEngine.Pool;
+#else
+using UnityEngine.SceneManagement;
+using System.Collections.Generic;
+#endif
+
+namespace Perigon.Utility
+{
+    public class ObjectPooler<T> where T : Component
+    {
+		protected string name;
+
+#if UNITY_2021_1_OR_NEWER
```

Comment:
So Instead of having a bunch of these everywhere, can you have 1 `bool flag` property that get's set as true within this, and then you can do all your code within checks of that flag.

Reply:
I'm not sure that will work, the UnityEngine.Pool api doesn't exist in sub 2021 versions so wrapping it in bools won't work without compiler conditionals

Reply:
ohh i see. Good to know.

Reply:
https://perigongames.atlassian.net/browse/BFB-127 

Made a ticket to refactor as reminder if/when we get to 2021